File: Algorithms/Medium/rotate_array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        
        const size_t dim = nums.size();
        vector<int> result(dim);

        for (size_t i = 0; i != dim; ++i) {
            if (i + k < dim) {
                result[i + k] = nums[i];
            } else {
                result[(i + k) % dim] = nums[i];
            }
        }

        for (size_t i = 0; i != dim; ++i) {
            nums[i] = result[i];
        }
    }
};

File: Algorithms/Medium/rotate_array.py
from typing import List


class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        res = [0] * len(nums)

        i = 0
        while i < len(nums):
            try:
                res[i + k] = nums[i]
            except IndexError:
                res[(i + k) % len(nums)] = nums[i]
            i += 1

        i = 0
        while i < len(nums):
            nums[i] = res[i]
            i += 1
